crates/sncast/tests/integration/data_transformer.rs

```diff
@@ -0,0 +1,1022 @@
+        &input,
+        contract_class,
+        &get_selector_from_name("enum_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0x1"),
+        Felt::from_hex_unchecked("0x80"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_enum_function_one_argument_variant_serialized_input() -> anyhow::Result<()>
+{
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("0x1"), String::from("0x80")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("enum_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0x1"),
+        Felt::from_hex_unchecked("0x80"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_enum_function_nested_struct_variant_cairo_expression_input(
+) -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from(
+        "Enum::Three(NestedStructWithField { a: SimpleStruct { a: 123 }, b: 234 })",
+    )];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("enum_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0x2"),
+        Felt::from_hex_unchecked("0x7b"),
+        Felt::from_hex_unchecked("0xea"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_enum_function_nested_struct_variant_serialized_input() -> anyhow::Result<()>
+{
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![
+        String::from("0x2"),
+        String::from("0x7b"),
+        String::from("0xea"),
+    ];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("enum_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0x2"),
+        Felt::from_hex_unchecked("0x7b"),
+        Felt::from_hex_unchecked("0xea"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_enum_function_invalid_variant_cairo_expression_input() {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("Enum::InvalidVariant")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("enum_fn").unwrap(),
+    );
+
+    result
+        .unwrap_err()
+        .assert_contains(r#"Couldn't find variant "InvalidVariant" in enum "Enum""#);
+}
+
+#[tokio::test]
+async fn test_happy_case_complex_struct_function_cairo_expression_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let data = concat!(
```

Comment:
use `indoc!`

Reply:
Good point, didn't think it will work here :))